nuget/lib/dependabot/nuget/file_updater.rb

```diff
@@ -45,7 +59,7 @@ def updated_dependency_files
             normalized_content = normalize_content(f, updated_content)
             next if normalized_content == f.content
 
-            next if only_deleted_lines?(f.content, normalized_content)
+            next unless FileUpdater.differs_in_more_than_blank_lines?(f.content, normalized_content)
```

Comment:
Does the original scenario (https://github.com/dependabot/dependabot-core/pull/9162) still pass here?

> If the updated content has less lines than the original content, then a package has been removed somewhere and we likely don't want to create a pull request.

Maybe there needs to be a test for this as well, unless there are other checks elsewhere to deal with this problem now.

Reply:
Good call, that was a test case I missed.  I just pushed a commit that adds this test case if you want to double check I understood the issue.

Reply:
Re-reading your comment I think there is a separate issue to address; what if a package is removed?  I think that should be handled in the C# code that actually checks the file content.  This part of the code is to handle silliness around empty lines.  There's likely a bigger work item to ensure that a dependency isn't removed, but we have better unit tests in the C# code for that.  Do you have a scenario where a package was removed?

Reply:
My understanding of the description in https://github.com/dependabot/dependabot-core/pull/9162 is that if you had a project like this:
```csproj
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net461</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="DependencyA" Version="1.0.0" />
  </ItemGroup>
</Project>
```
And NuGetUpdater attempted to update "DependencyA" to "2.0.0" but failed and rolled back e.g. 
```log
WARNING: Install failed. Rolling back...
```

It would remove the dependency entirely and result in a project file like:
```diff
~~~
@@ -2,7 +2,4 @@
   <PropertyGroup>
     <TargetFramework>net461</TargetFramework>
   </PropertyGroup>
-  <ItemGroup>
-    <PackageReference Include="DependencyA" Version="1.0.0" />
-  </ItemGroup>
 </Project>
\ No newline at end of file
```

In this scenario, the file is different, but it is different in a way that the PR should not be created because the change isn't desired. I could be misunderstanding this problem though.

Maybe the problem shouldn't be solved in `file_updater` and instead the "rollback" warning should be escalated as an error by the native NuGet updater?

I'll let you decide what is correct here as you understand this all better than me. 
Ultimately this change solves my specific problem, so I'm happy with it.

Reply:
I'd say the scenario you presented should be handled purely in C#.  Let's take this PR as is, because it does fix a class of errors that we're currently seeing and I'll look at the rollback scenario tomorrow and see if I can nail down what might cause that.  There appears to be enough information in the linked issue to sort it out.

Reply:
@rhyskoedijk I filed issue #10342 to track this other scenario and I'm working on it now.  Rough ETA, sometime next week.  I think the best mechanism to prevent this kind of error is to only report available package upgrades if all transitive dependencies can be resolved so that we never try to update when it's not fully resolvable (the base scenario that #9162 was trying to address).